### ISSUE TITLE
An empty "rules" array scans every file and exits 0 silently, while every other zero-rule config is an error

### ISSUE THREAD

**Issue by konard** [secretlint/secretlint#1688]
## Summary

A `.secretlintrc.json` whose `rules` array is **empty** makes secretlint scan every file,
report nothing and exit `0` — with no warning that it ran with zero rules loaded.

That makes "configured with no rules" indistinguishable from "the tree is clean", which is
the one thing a secret scanner must never be ambiguous about. It is also inconsistent with
secretlint's own behaviour on every other misconfiguration: the three neighbouring mistakes
are all hard errors.

## Reproduction

secretlint 13.0.5, Node v22.23.2 (`node:22-slim`):

```console
$ mkdir /w && cd /w
$ KEY="AKIA$(LC_ALL=C tr -dc 'A-Za-z0-9' </dev/urandom | head -c 16)"
$ SEC="$(LC_ALL=C tr -dc 'A-Za-z0-9' </dev/urandom | head -c 40)"
$ printf 'aws_access_key_id = %s\naws_secret_access_key = %s\n' "$KEY" "$SEC" > canary.txt
```

**Control — the file is detectable:**

```console
$ echo '{ "rules": [ { "id": "@secretlint/secretlint-rule-preset-recommend" } ] }' > .secretlintrc.json
$ npx -p secretlint@13.0.5 -p @secretlint/secretlint-rule-preset-recommend secretlint canary.txt
/w/canary.txt
  2:0  error  [AWSSecretAccessKey] found AWS Secret Access Key: ****  @secretlint/secretlint-rule-preset-recommend > @secretlint/secretlint-rule-aws

✖ 1 problem (1 error, 0 warnings, 0 infos)
$ echo $?
1
```

**The report — same file, empty `rules`:**

```console
$ echo '{ "rules": [] }' > .secretlintrc.json
$ npx -p secretlint@13.0.5 -p @secretlint/secretlint-rule-preset-recommend secretlint canary.txt
$ echo $?
0
```

No output at all. `--format json` is equally indistinguishable from a clean scan:

```console
$ npx ... secretlint --format json canary.txt
[{"filePath":"/w/canary.txt","sourceContent":"aws_access_key_id = AKIA…","sourceContentType":"text","messages":[]}]
$ echo $?
0
```

## Why this is inconsistent

Every *other* way of ending up with no rules is already a hard error:

| `.secretlintrc.json` | Behaviour |
| --- | --- |
| file absent | `Error: secretlint config is not found` — exit 2 |
| `{ "rule": [ … ] }` (typo) | `Error: secretlintrc should have required 'rules' property.` — exit 2 |
| `{ "rules": [ { "id": "@secretlint/secretlint-rule-does-not-exist" } ] }` | `AggregationError: … Failed to load rule module` — exit 2 |
| **`{ "rules": [] }`** | **silent, exit 0** |

The existing messages say the config file "define the use of rules", so a config that
defines none is arguably already outside what the format means.

Realistically this is reached by a tool that generates the config, by commenting entries
out while debugging, or by a merge that empties the array — none of which announce
themselves, because the scan stays green.

## Workaround

Validate the scanner before trusting its silence: plant a secret, scan it in the same
invocation, and fail if it is *not* reported.

```sh
CANARY_DIR="$(mktemp -d)"; trap 'rm -rf "$CANARY_DIR"' EXIT
cp .secretlintrc.json "$CANARY_DIR/"
rand() { LC_ALL=C tr -dc 'A-Za-z0-9' </dev/urandom | head -c "$1"; }
{ printf 'aws_access_key_id = AKIA%s\n' "$(rand 16)"
  printf 'aws_secret_access_key = %s\n' "$(rand 40)"; } > "$CANARY_DIR/canary.txt"

if (cd "$CANARY_DIR" && secretlint canary.txt) >/dev/null 2>&1; then
  echo "secretlint did not flag a planted key: the rules are not live" >&2
  exit 1
fi
secretlint "**/*"
```

(The canary is generated rather than written down, so the script is not itself a finding,
and so it can never quietly become an allow-listed constant. Detection with this generator
was 100/100 over 100 samples with `preset-recommend`.)

## Suggested fix

The check is one condition next to the one that already exists, in
`packages/@secretlint/config-loader/src/validator.ts` → `validateConfigDescriptor`:

```ts
if (!Array.isArray(configDescriptor.rules)) {
    return { ok: false, error: new Error(`secretlintrc should have required 'rules' property. …`) };
}
+ if (configDescriptor.rules.length === 0) {
+     return {
+         ok: false,
+         error: new Error(
+             `secretlintrc has an empty "rules" array, so no rule would run and every file would pass.\n` +
+             `Add at least one rule or preset, for example:\n\n` +
+             `{\n    "rules": [\n        { "id": "@secretlint/secretlint-rule-preset-recommend" }\n    ]\n}\n`
+         ),
+     };
+ }
```

If erroring is considered too strict (someone may deliberately want a no-op config), the
two softer options both remove the ambiguity:

* emit a warning on stderr — `warning: no rules are configured; every file will pass` —
  while keeping exit `0`; or
* exit non-zero only under a flag such as `--fail-on-no-rules`, so CI can opt in.

I would be glad to send a PR for whichever of the three you prefer.

## Context

Found while auditing CI for [link-foundation/box](https://github.com/link-foundation/box)
(issue [#115](https://github.com/link-foundation/box/issues/115)), whose theme is exactly
this: checks whose green is indistinguishable from "the check did not run".
